api/src/main/java/net/kyori/adventure/audience/Audience.java

```diff
@@ -58,129 +101,125 @@
       return empty();
     } else if(length == 1) {
       return audiences[0];
+    } else {
+      return of(Arrays.asList(audiences));
     }
-    return of(Arrays.asList(audiences));
   }
 
   /**
-   * Creates an audience that delegates to a collection of audiences.
+   * Creates an audience that forwards to many other audiences.
    *
-   * @param audiences the delegate audiences
+   * <p>The underlying <tt>Iterable</tt> is not copied, therefore any changes
+   * made will be reflected in <tt>Audience</tt>.</p>
+   *
+   * @since 4.0.0
+   * @see ForwardingAudience
+   * @param audiences an iterable of audiences, can be empty
    * @return an audience
    */
   static @NonNull Audience of(final @NonNull Iterable<? extends Audience> audiences) {
-    return (MultiAudience) () -> audiences;
+    return (ForwardingAudience) audiences;
```

Comment:
```suggestion
    return (ForwardingAudience) () -> audiences;
```

Reply:
Curious what the difference is here.

Reply:
one is a cast, one creates an instance of the functional interface with an explicit type hint

Reply:
yes, the `() ->` is to create a method handle that returns the local var on call